static/js/components/widgets/HierarchicalSelectField.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState } from "react"
+import { sortedUniqBy, sortBy, times } from "lodash"
+
+import SelectField from "./SelectField"
+
+type Level = { label: string; name: string }
+type Option = { label: string; value: string }
+type OptionsMap = Record<string, Record<string, any> | string[]>
+type Props = {
+  name: string
+  label: string
+  label_singular?: string // eslint-disable-line camelcase
+  levels: Level[] // eslint-disable-line camelcase
+  value: Tuple[] | null
+  onChange: (event: any) => void
+  options_map: OptionsMap // eslint-disable-line camelcase
+}
+
+const asOption = (str: string | null) => ({
+  label: str ?? "-- empty --",
+  value: str ?? ""
+})
+
+type TupleItem = string | null
+type Tuple = TupleItem[]
```

Comment:
what would you think about maybe using a different name for this? I was thinking possibly it could be a `TopicTaxonomy` or `TopicSet` or something along those lines? `Tuple` is just a wee bit of a generic name I think